test/unit/double_free.c

```diff
@@ -30,8 +31,14 @@ TEST_BEGIN(test_large_double_free_tcache) {
 
 	test_large_double_free_pre();
 	char *ptr = malloc(SC_LARGE_MINCLASS);
+	bool guarded = extent_is_guarded(tsdn_fetch(), ptr);
 	free(ptr);
-	free(ptr);
+	if (!guarded) {
+		free(ptr);
+	} else {
+		/* Similarly, skip guarded ones. */
+		fake_abort_called = true;
```

Comment:
Why do we need to skip guarded allocations?

Reply:
Actually I just realized we only need to skip the `_no_tcache` test, in which case, the guarded allocation will have immediate actions (unguard for large) that crashes before the intended sanity check.

For the tcache enabled case, the large tcache sized allocations can sneak through even with double frees, allowing the test to reach into the tcache_flush which does the sanity checks.

Will update and add comments.